velodrome/token-counter/token-counter.go

```diff
@@ -23,24 +23,64 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"net/http"
 	"os"
 	"path/filepath"
 	"strings"
 	"time"
 
 	"github.com/golang/glog"
 	"github.com/google/go-github/github"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"golang.org/x/oauth2"
 )
 
 type tokenCounterFlags struct {
-	influx InfluxConfig
-	tokens []string
+	influx      InfluxConfig
+	tokens      []string
+	metricsPort int
+	host        string
+	promOnly    bool
+}
+
+func NewCollector(tokens []TokenHandler) prometheus.Collector {
+	return &Collector{
+		tokens: tokens,
+		tokenCounter: prometheus.NewDesc(
+			"github_token_count",
+			"Number of counted API calls against the github API within a reset window",
+			[]string{"login"}, nil),
```

Comment:
@spiffxp just came to my mind that it might make sense to add a `rate` label too. Alongside to `login`, `rate` could then have the dimension `remaining` and `used`. Having `remaining` might be more correct in the case where a token is actually blocked, because you would then see `0` for `remaining` instead of `0` used, which could also mean that no requests to the api occurred. What do you think?

Reply:
Sounds good, you're welcome to amend this PR or open a followup

Reply:
Added it to the PR. Thanks.